### PR TITLE
added external sync mode and shared config for head and sensorrring

### PR DESCRIPTION
**Pull request by mathias-luedtke**
This config sets up an external sync (needs https://github.com/ros-industrial/ros_canopen/pull/201) from a shared config.

alternative to #566

### CODE REVIEW

**Comment by mathias-luedtke** [comment]
This is my preferred solution.

**Review by fmessmer — CHANGES_REQUESTED** [review]
Travis fails:

```
-------------------------------------------------

Full test results for 'build/test_results/cob_bringup/roslaunch-check_cob4-1__ROBOT_ENV_ipa-kitchen.xml.xml'

-------------------------------------------------

<?xml version="1.0" encoding="UTF-8"?>

<testsuite tests="1" failures="1" time="1" errors="0" name="roslaunch-check_cob4-1__ROBOT_ENV_ipa-kitchen.xml.xml">

  <testcase name="test_ran" status="run" time="1" classname="Results">

  <failure message="roslaunch check [/home/travis/catkin_ws/src/cob_robots/cob_bringup/robots/cob4-1.xml] failed" type=""/>

  </testcase>

  <system-out><![CDATA[[

[/home/travis/catkin_ws/src/cob_robots/cob_bringup/robots/cob4-1.xml]:

	Missing package dependencies: cob_bringup/package.xml: canopen_chain_node

cannot find node [canopen_sync_node] in package [canopen_chain_node]

]]></system-out>

</testsuite>
```

File: cob_bringup/drivers/canopen_402.launch [hunk 1]
```diff
@@ -5,11 +5,23 @@
 	<arg name="pkg_hardware_config" default="$(find cob_hardware_config)"/>
 	<arg name="component_name"/>
 	<arg name="can_device"/>
+	<arg name="interval_ms" default="10"/>
+	<arg name="overflow" default="0"/>
+	<arg name="use_external_sync" default="false"/>
+	<arg name="start_external_sync" default="$(arg use_external_sync)"/>
+	
+	<!-- start external sync node if requested -->
+	<node pkg="canopen_chain_node" type="canopen_sync_node" name="sync_$(arg can_device)_$(arg component_name)" output="screen" if="$(arg start_external_sync)">
```
Comment (fmessmer) [inline cob_bringup/drivers/canopen_402.launch:14]: please add dependency to  ```canopen_chain_node``` to cob_bringup/package.xml

**Review by fmessmer — CHANGES_REQUESTED** [review]
Travis fails:

```
-------------------------------------------------

Full test results for 'build/test_results/cob_bringup/roslaunch-check_cob4-1__ROBOT_ENV_ipa-kitchen.xml.xml'

-------------------------------------------------

<?xml version="1.0" encoding="UTF-8"?>

<testsuite tests="1" failures="1" time="1" errors="0" name="roslaunch-check_cob4-1__ROBOT_ENV_ipa-kitchen.xml.xml">

  <testcase name="test_ran" status="run" time="1" classname="Results">

  <failure message="roslaunch check [/home/travis/catkin_ws/src/cob_robots/cob_bringup/robots/cob4-1.xml] failed" type=""/>

  </testcase>

  <system-out><![CDATA[[

[/home/travis/catkin_ws/src/cob_robots/cob_bringup/robots/cob4-1.xml]:

	Missing package dependencies: cob_bringup/package.xml: canopen_chain_node

cannot find node [canopen_sync_node] in package [canopen_chain_node]

]]></system-out>

</testsuite>
```

File: cob_bringup/drivers/canopen_402.launch (hunk 1 above)
Comment (mathias-luedtke) [reply cob_bringup/drivers/canopen_402.launch:14]: (`canopen_chain_node` is a dependency of `canopen_motor_node` thus)
`canopen_sync_node` will be available with https://github.com/ros-industrial/ros_canopen/pull/201)

**Comment by mathias-luedtke** [comment]
added dependency.

**Review by fmessmer — APPROVED** [review]

File: cob_bringup/drivers/canopen_402.launch (hunk 1 above)
Comment (fmessmer) [reply cob_bringup/drivers/canopen_402.launch:14]: https://github.com/ipa320/cob_robots/pull/568/files#diff-cca74cba52fdd3b91eb2f4a7e4b67816R19

**Comment by mathias-luedtke** [comment]
2d90eb018304206f7c91c9b2f8f2c43bc089e717 should be reverted later:
* remove can settings from config files
* set sync rate in launch file

In addition `use_external_sync=true` should become the new default.

**Comment by fmessmer** [comment]
I'm trying to understand which of the related PRs are still relevant...
It seems that this PR replaces #561 and #566...is this correct?
Any changes from one of those PRs that needs to be added to this one?
What about [your comment](https://github.com/ipa320/cob_robots/pull/568#issuecomment-267600358)? Can this be finalized?
I guess travis fails because latest ros_canopen (including external sync) is not yet released....

**Comment by mathias-luedtke** [comment]
I have closed the others PRs.
My comment should become a new issue to keept track of it.

> I guess travis fails because latest ros_canopen (including external sync) is not yet released....

It is not released as it should be tested a little bit longer ;)
Just keep me posted for planned `cob_robots` releases.

**Comment by fmessmer** [comment]
@ipa-mdl @ipa-nhg 
So do we need this for cob4-10/cob4-7? 
If we do not merge this how are head and sensorring actuated on hardware?

**Comment by floweisshardt** [comment]
included in https://github.com/ipa320/cob_robots/pull/593 which was tested successfully on cob4-7/cob4-10

**Comment by ipa-fxm** [comment]
I put a note about high load problem caused by the `cartesian_control` [here](https://github.com/ipa320/cob_control/issues/101#issuecomment-274008953)...will work on this as soon as I find the time

**Pull request merged**